### DISCUSSION TITLE
Trying to call a dll (converted from a jar) in a web application, getting "Could not locate ikvm.home when establishing VFS".

### DISCUSSION THREAD

**Discussion by apn-hroussel** [ikvmnet/ikvm#356] [Q&A] [answered]
OK.  So following my success with a jar converted to an executable, I tried to convert a jar to a dll and then use this dll as part of a service handler (web app running under IIS).

I got the "Could not locate ikvm.home when establishing VFS" message, even though I had a copy of the "ikvm" runtime under the "bin" directory of my app.  I even confirmed, with procmon64.exe, that the directories were found and visited.

Is there anything that would process differently, from the point of view of ikvm, because we're running under IIS?

Has anybody else tried to convert a jar to a dll and load that from an aspx page, or a web service under IIS?

Thanks.

**Comment by wasabii** [comment]
Shadow copies are the issue.

**Reply by apn-hroussel** [reply, marked as answer]
Thanks, that worked!

For anybody else who has the same issue, you need to put,

<hostingEnvironment shadowCopyBinAssemblies="false" />

in the <system.web> section of your web.config.

**Reply by Prabhat070** [reply]
I can t see what to add under system.web except comma

**Reply by Prabhat070** [reply]
What to add. Pls help..its important

**Reply by apn-hroussel** [reply]
See my answer below.  Original line seemed to have gotten erased because it was html markup maybe?

**Comment by Prabhat070** [comment]
> Thanks, that worked!
> 
> For anybody else who has the same issue, you need to put,
> 
> in the <system.web> section of your web.config.

**Comment by kc6ng** [comment]
May I know what should I put into web.config?

**Reply by apn-hroussel** [reply]
OK my bad I didn't realize the most important line had gotten erased.

I needed to put,

&lt;hostingEnvironment shadowCopyBinAssemblies="false" /&gt;

**Reply by kc6ng** [reply]
Thanks. It works for my project.
